loopy/target/cuda.py

```diff
@@ -299,6 +291,16 @@ def cuda_preamble_generator(preamble_info):
             #endif
             """)
 
+    from loopy.tools import remove_common_indentation
+    kernel = preamble_info.kernel
+    idx_nbits = kernel.index_dtype.itemsize * 8
+    idx_ctype = kernel.target.dtype_to_typename(kernel.index_dtype)
+    yield (f"00_declare_gid_lid_{idx_nbits}",
+           remove_common_indentation(f"""
+                #define bIdx_{idx_nbits}(N) (({idx_ctype}) blockIdx.N)
+                #define tIdx_{idx_nbits}(N) (({idx_ctype}) threadIdx.N)
```

Comment:
If the goal is to decrease SNR, why include the bit count?

Reply:
I thought it was necessary when there are non-entrypoint kernels that have an index dtype that is not equal to that of the entrypoint kernel, as we might need different casts. Maybe I could do `bIdx` for `i32` and `bIdx_64` for `i64`?

Reply:
Hmm. I feel like we should instead check that everybody agrees on the index type. Given that inames get passed between kernels, I foresee trouble if there isn't agreement.

Reply:
Otherwise, what you point out is a bug that ought to get fixed in CL as well. What do you think?

Reply:
Agree with both. I don't see a good practical reason to use different index dtypes for kernels reachable from a single entrypoint. I'll patch it!